packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -93,6 +95,10 @@ export class PropertiesManager {
                     this.pendingKeyUpdateCount[key]++;
                 } else if (!shouldModifyKey(key)) {
                     continue;
+                } else if (seq === UniversalSequenceNumber
+                    && this.pendingKeyUpdateCount?.[key] !== undefined && this.pendingKeyUpdateCount?.[key] > 0) {
+                        // decrement if local annotation is rolled back
+                        this.pendingKeyUpdateCount[key]--;
```

Comment:
Using UniversalSequenceNumber to decide to decrement seems wrong since it doesn't necessarily mean anything about rollback. I could pass a parameter to indicate rollback or add another method to clean up pendingKeyUpdateCount values.

(pendingRewriteCount also needs to be decremented, but I'm waiting to do that when I have the right solution for pendingKeyUpdateCount.)

Reply:
Yeah, it does seem weird. I'm not sure it causes incorrect behavior, but having clearer semantics seems better. I think passing parameter to indicate rollback or having a separate rollback method on `SegmentPropertiesManager` is better than exposing internals of pendingKeyUpdateCount (even if indirectly).